sdk/__init__.py

```diff
@@ -76,6 +80,10 @@ def create_batch(
         if device_type != DeviceType.QPU:
             req.update({"emulator": device_type})
 
+        # The configuration field is only added in the case
+        # it's requested
+        if configuration:
+            req.update({"configuration": configuration.to_dict()})
         batch_rsp, jobs_rsp = self._client._send_batch(req)
         batch = Batch(**batch_rsp, _client=self._client)
```

Comment:
The result of this deserialization will lead the configuration attribute to be a dict and not a Configuration instance unless dataclasses operate some magic I'm not aware of, but it doesn't look like it in my experiments:

```bash
>>> @dataclass
... class Yolo:
...   id: int = 0
...   configuration: Configuration = None
... 
>>> 
>>> 
>>> 
>>> Yolo(**{"id": 1})
Yolo(id=1, configuration=None)
>>> Yolo(**{"id": 1, "configuration": { "dt" : 0.1}})
Yolo(id=1, configuration={'dt': 0.1})
```

With the current implementation of the configuration dataclass, I don't really see how we are going to circumvent that issue tbh. All the dict fields will be on the same level in the response of the core API and we need to separate the `extra_config` fields from the default value fields to build the configuration object.

Reply:
that was the whole point :) 

With a growing number of emulators it will be much more manageable to have them on the first level downstream.

Suppose in the future we'll have 8 different versions of the emulator, all of which will have some number of custom options and others will be shared by an arbitrary number of emulators.

To some extent, the distinction between free form and default config options are very minor, and not confirmed. Indeed, different emulators will have different default config options.

The current implementation is very hindered by the fact that this is a public repo and I can only show the parts that are for products already announced.